packages/multichain-network-controller/src/MultichainNetworkController.ts

```diff
@@ -137,6 +139,52 @@ export class MultichainNetworkController extends BaseController<
     return await this.#setActiveEvmNetwork(id);
   }
 
+  /**
+   * Removes an EVM network from the list of networks.
+   * This method re-directs the request to the network-controller.
+   *
+   * @param chainId - The chain ID of the network to remove.
+   * @returns - A promise that resolves when the network is removed.
+   */
+  async #removeEvmNetwork(chainId: CaipChainId): Promise<void> {
+    const hexChainId = convertCaipToHexChainId(chainId);
+    const selectedChainId = this.messagingSystem.call(
+      'NetworkController:getSelectedChainId',
+    );
+
+    if (this.state.isEvmSelected && selectedChainId === hexChainId) {
+      throw new Error('Cannot remove the currently selected network');
+    }
+
+    if (!this.state.isEvmSelected && selectedChainId === hexChainId) {
+      const ethereumMainnetHexChainId = '0x1';
+      const clientId = this.messagingSystem.call(
+        'NetworkController:findNetworkClientIdByChainId',
+        ethereumMainnetHexChainId,
+      );
+
+      await this.messagingSystem.call(
+        'NetworkController:setActiveNetwork',
+        clientId,
+      );
+    }
+
+    this.messagingSystem.call('NetworkController:removeNetwork', hexChainId);
+  }
+
+  /**
+   * Removes a network from the list of networks.
+   * It only supports EVM networks.
+   *
+   * @param chainId - The chain ID of the network to remove.
+   * @returns - A promise that resolves when the network is removed.
+   */
+  async removeNetwork(chainId: CaipChainId): Promise<void> {
+    if (isEvmCaipChainId(chainId)) {
+      await this.#removeEvmNetwork(chainId);
+    }
+  }
```

Comment:
I wonder if we should throw if for non-EVM networks 🤔

Like, if someone really tried to remove a non-EVM network (programmatically speaking), having an error could be better than just doing "nothing", WDYT?

Reply:
wdyt https://github.com/MetaMask/core/pull/5516/commits/240497945c87ba4525856ba3b346d594f6100840?

Reply:
Perfect! thanks!